### ISSUE TITLE
bug(makefile) - make install fails on Ubuntu/Salt/Py3

### ISSUE THREAD

**Issue by noelmcloughlin** [SUSE/DeepSea#1702]
### Description of Issue/Question

TL;DR; The `DEEPSEA_DEPS=` variable must be OS specific.

Running [salt-bootstrap with python3](https://github.com/saltstack/salt-bootstrap#python-3-support) configures the correct repo; and salt-api is running.
```
root@ubuntu1804:/root/Deepsea# head /etc/apt/sources.list.d/saltstack.list 
deb https://repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/2019.2.0 bionic ..

root@ubuntu1804:/etc/apt# head -1 /usr/bin/salt-api
#!/usr/bin/python3
```
but `make install` fails on Ubuntu.
```
See "systemctl status salt-api.service" and "journalctl -xe" for details.
Makefile:974: recipe for target 'install' failed
make: *** [install] Error 1
```
This was a puzzle but I traced the problem to  statement `$(PKG_INSTALL) $(DEEPSEA_DEPS) || true` discussed in PR #1140, #1196, and #1285 which bizarrely triggers a python2 installation?
```
apt-get install -y salt-api || true
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libpython-stdlib libpython2.7-minimal libpython2.7-stdlib libssl1.1 python python-asn1crypto python-cffi-backend python-cherrypy3
  python-cryptography python-enum34 python-idna python-ipaddress python-minimal python-openssl python-pkg-resources python-repoze.lru
  python-routes python-simplejson python-six python-webob python2.7 python2.7-minimal
Suggested packages:
  python-doc python-tk python-cryptography-doc python-cryptography-vectors python-enum34-doc python-openssl-doc python-openssl-dbg
  python-setuptools python-paste python-webob-doc python2.7-doc binutils binfmt-support
The following NEW packages will be installed:
  libpython-stdlib libpython2.7-minimal libpython2.7-stdlib python python-asn1crypto python-cffi-backend python-cherrypy3
  python-cryptography python-enum34 python-idna python-ipaddress python-minimal python-openssl python-pkg-resources python-repoze.lru
  python-routes python-simplejson python-six python-webob python2.7 python2.7-minimal
The following packages will be upgraded:
  libssl1.1 salt-api
2 upgraded, 21 newly installed, 0 to remove and 155 not upgraded.
```
And the wrong repo was configured by ubuntu.
```
root@ubuntu1804:/root/Deepsea# head /etc/apt/sources.list.d/saltstack.list 
deb http://repo.saltstack.com/apt/ubuntu/18.04/amd64/latest bionic main
```

And `salt-api` package is wrong.
```
root@ubuntu1804:/etc/apt# head -1 /usr/bin/salt-api
#! /usr/bin/python2
```

### Workaround

I started from scratch and set `DEEPSEA_DEPS=''` in the Makefile before `make install`. 
This resolved the problem.
```
apt-get install -y  || true
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 157 not upgraded.
apt-get install -y python3-setuptools python3-click tox python3-configobj || true
Reading package lists... Done
     <<CUT>>
# Use '|| true' to suppress some error output in corner cases
systemctl restart salt-master
systemctl restart salt-api
# deepsea-cli
python3 setup.py install --root=/
    << CUT >>
running install_scripts
Installing deepsea script to /usr/local/bin
```
And `salt-api` package is correct.
```
root@ubuntu1804:/etc/apt# head -1 /usr/bin/salt-api
#!/usr/bin/python3
```

### Setup

This is some Ubuntu salt package (handling) nuance?

### Steps to Reproduce Issue

Install Ubuntu 18.04
Install Salt per https://github.com/saltstack/salt-bootstrap#python-3-support
`git clone https://github.com/SUSE/DeepSea.git
cd DeepSea/`
 Update  Makefile per https://github.com/SUSE/DeepSea/pull/1697/files
`make install`

### Versions Report

Note this is situation after `make install` fails.
```
root@ubuntu1804:/root/Deepsea# salt-run deepsea.version
0.9.4+792.g3afb8e4a
root@ubuntu1804:/root/Deepsea# dpkg-query --list |grep salt
ii  salt-api                              2019.2.0+ds-1                     all          Generic, modular network access system
ii  salt-common                           2019.2.0+ds-1                     all          shared libraries that salt requires for all packages
ii  salt-master                           2019.2.0+ds-1                     all          remote manager to administer servers via salt
ii  salt-minion                           2019.2.0+ds-1                     all          client package for salt, the distributed remote execution system
```

**Comment by noelmcloughlin** [comment]
I am willing to update #1697 with an agreed solution.

**Comment by noelmcloughlin** [comment]
Resolved by #1697 (probably).
